tools/koch/koch.nim

```diff
@@ -152,14 +168,87 @@ proc getSourceMetadata(): tuple[hash, date: string] =
       if dateCall.exitCode == 0:
         date = dateCall.output.strip()
 
-  result = (hash, date)
+      let nearestReleaseTagCall = execCmdEx:
+        "git" & " -C " & quoteShell(nimSource) & " describe" &
+        # Match all tags
+        " --tags" &
+        # Match versioned releases
+        " --match " & quoteShell"*.*.*" &
+        # Excluding pre-releases
+        " --exclude " & quoteShell"*-*"
+
+      # If there is a tag nearby
+      if nearestReleaseTagCall.exitCode == 0:
+        # If `-` is not in the tag, it means the tag is of the same commit as
+        # what is being compiled.
+        if '-' notin nearestReleaseTagCall.output:
+          let taggedVersion = nearestReleaseTagCall.output.strip()
+
+          if taggedVersion != VersionAsString:
+            quit:
+              "The compiler version (" & VersionAsString & ")" &
+              " is different from the tagged release: " & taggedVersion
+
+          versionSuffix = ""
+
+        # If `-` is in the tag, then the tag is of a previous release.
+        else:
+          let
+            # Git produces the following format: <tag>-<distance>-g<short commit id>
+            #
+            # We are interested in the tag and the commit distance.
+            splittedOutput = nearestReleaseTagCall.output.strip().rsplit('-')
```

Comment:
```suggestion
            splitOutput = nearestReleaseTagCall.output.strip().rsplit('-')
```
Splitted isn't a word, but it also is a bit more clear that it's past tense, grammar objections aside. I can go either way.